src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -702,24 +714,14 @@ def __call__(
                     if callback is not None and i % callback_steps == 0:
                         callback(i, t, latents)
 
-        if output_type == "latent":
-            image = latents
-            has_nsfw_concept = None
-        elif output_type == "pil":
-            # 8. Post-processing
-            image = self.decode_latents(latents)
-
-            # 9. Run safety checker
+        if not output_type == "latent":
+            image = self.vae.decode(latents / self.vae.config.scaling_factor).sample
             image, has_nsfw_concept = self.run_safety_checker(image, device, prompt_embeds.dtype)
-
-            # 10. Convert to PIL
-            image = self.numpy_to_pil(image)
         else:
-            # 8. Post-processing
-            image = self.decode_latents(latents)
-
-            # 9. Run safety checker
-            image, has_nsfw_concept = self.run_safety_checker(image, device, prompt_embeds.dtype)
+            has_nsfw_concept = False
```

Comment:
I believe this is a bug from previous versions. According to the documentation:

https://github.com/huggingface/diffusers/blob/7c1bb9ae81c97f6f10cf36784c17a51ebddeea08/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L628-L631

and

https://github.com/huggingface/diffusers/blob/7c1bb9ae81c97f6f10cf36784c17a51ebddeea08/src/diffusers/pipelines/stable_diffusion/__init__.py#L30-L32

we are supposed to always return a list. Therefore we'd need to do something like this, same thing in `run_safety_checker` and then simplify `do_normalize` accordingly:

```suggestion
            has_nsfw_concept = [False] * latents.shape[0]
```